### PR TITLE
refactor(vision): migrate ten classifier families

### PR DESCRIPTION
**Pull request by yifeif-nv**
## Background

Migrate ten remaining timm classifier families to the existing Task SDK:
DPN, HRNet, MNASNet, NFNet, RepVGG, Res2Net, SENet, SE-ResNet, VGG and Xception.
All changes stay in their owning `families/timm_*/` directories. This continues
the merged classifier migrations without adding a shared model implementation.

## Exit Criteria

- Each family implements `IModel` and `IImageToClassScores`, binds its own Task,
  and returns every class logit in checkpoint order.
- Preserve model computation, preprocessing, declared profiles and existing
  acceptance rules; validate the original selected checkpoints and public SDK consumers.
- Required remote checks must pass before merge. This submission does not claim
  current-head Pre-merge success.

## Implementation

- Change owned support/build/manifests to `image_to_class_scores`; return complete
  owned scores plus explicit checkpoint labels/vocabulary, or empty identity when unknown.
- Keep graph, weights, preprocessing implementations and original reference
  paths unchanged, including NFNet's center/squash crop policy.
- Add family-local CPU binding/metadata/lifetime tests and public C11/C++17
  consumers. Existing E2Es keep the JPEG CLI path and original top-1 oracle;
  SDK consumers receive identical decoded RGB and compare all scores.
- Wire consumers through each family's existing CTest dependencies. Take over
  existing performance entry IDs through the family-owned reference protocol,
  retaining workloads, precision, 3 warmups, 10 measurements and 5% margin.
- No Core, shared header, registry, CLI, workflow, dependency or sibling-family changes.

### Change categories

- [x] Model or runtime behavior
- [x] Public API
- [ ] ABI
- [x] Bundle or artifact format
- [ ] Dependencies
- [ ] Documentation only
- [x] CI or developer tooling

Public C layouts and bundle format 1 are unchanged. Family primary Task and runtime
metadata change; rebuild existing bundles. No legacy family interface or fallback is added.

### Commands and Results

With the existing native build/E2E environment:

- `cmake --build /build --parallel 8 --target trtmc trtmc_backend_trt trtmc_benchmark_worker test_timm_mnasnet_task_contract test_timm_mnasnet_image_preprocess test_timm_dpn_task_contract test_timm_dpn_image_preprocess test_timm_hrnet_task_contract test_timm_hrnet_image_preprocess test_timm_nfnet_task_contract test_timm_nfnet_image_preprocess test_timm_repvgg_task_contract test_timm_repvgg_image_preprocess test_timm_res2net_task_contract test_timm_res2net_image_preprocess test_timm_senet_task_contract test_timm_senet_image_preprocess test_timm_seresnet_task_contract test_timm_seresnet_image_preprocess test_timm_vgg_task_contract test_timm_vgg_image_preprocess test_timm_xception_task_contract test_timm_xception_image_preprocess`: passed, including all twenty public SDK consumers.
- `ctest --test-dir /build --output-on-failure -R '^(timm_mnasnet|timm_dpn|timm_hrnet|timm_nfnet|timm_repvgg|timm_res2net|timm_senet|timm_seresnet|timm_vgg|timm_xception)_(task_contract|image_preprocess)$'`: 20/20 passed.
- `python -m pytest families/timm_mnasnet/tests families/timm_dpn/tests families/timm_hrnet/tests families/timm_nfnet/tests families/timm_repvgg/tests families/timm_res2net/tests families/timm_senet/tests families/timm_seresnet/tests families/timm_vgg/tests families/timm_xception/tests -q`: 308 passed; 14 explicitly unselected E2E cases skipped.
- `python -m pytest families/timm_mnasnet/tests/test_e2e.py families/timm_dpn/tests/test_e2e.py families/timm_hrnet/tests/test_e2e.py families/timm_nfnet/tests/test_e2e.py families/timm_repvgg/tests/test_e2e.py families/timm_res2net/tests/test_e2e.py families/timm_senet/tests/test_e2e.py families/timm_seresnet/tests/test_e2e.py families/timm_vgg/tests/test_e2e.py families/timm_xception/tests/test_e2e.py --e2e-testcase mnasnet-100-rmsp-in1k --e2e-testcase dpn68b-ra-in1k --e2e-testcase hrnet-w18-ms-aug-in1k --e2e-testcase dm-nfnet-f0-dm-in1k --e2e-testcase repvgg-a2-rvgg-in1k --e2e-testcase res2net50-26w-4s-in1k --e2e-testcase senet154-gluon-in1k --e2e-testcase seresnet50-a1-in1k --e2e-testcase vgg16-tv-in1k --e2e-testcase xception41-tf-in1k -q`: 10 actual checkpoint cases passed, including C/C++ calls, in 457.36 seconds.
- `PYTHONPATH=core/builder:apps/benchmark:. python3 -m tools.community_ci source-quality --base 051bd976f536c8b459c274d795e2842811536b8b`: legal headers, inventory, shared-native complexity, formatting and 287 contract tests passed.
- `git diff --check`: passed.

MNASNet was brought through a minimal real end-to-end path before expanding the
other nine families. All ten original benchmark workloads then executed using the
existing suite resolver, native worker, family-reference command builder and
output-contract checker. All ten output contracts passed; each native and reference
result retained 1000 scores. This is execution/output evidence, not speedup qualification.

### Hardware, Environment, and Revisions

Tested branch contents: `55481c16f3fd4e4dda7ab8a7d4b59dd34035c794`.
Base: `051bd976f536c8b459c274d795e2842811536b8b`.
Linux/aarch64, Python 3.12.3, TensorRT 11.1.0.106, CUDA 13.3.33,
PyTorch 2.12.0+cu130 and timm 1.0.28; one NVIDIA GB300 per case, original FP16 profiles.

| Checkpoint | Resolved tested revision |
| --- | --- |
| `timm/dpn68b.ra_in1k` | `fb49a617159bd2a0d6099742a6c57ef2d884dad9` |
| `timm/hrnet_w18.ms_aug_in1k` | `7e2c5583769f54514fd87e3ba9de408e33eaba0f` |
| `timm/mnasnet_100.rmsp_in1k` | `a30af72360c7a4871b0156cc11d7b767208273d7` |
| `timm/dm_nfnet_f0.dm_in1k` | `ca8e830961c2ebceffff7affff34acfe7a1da673` |
| `timm/repvgg_a2.rvgg_in1k` | `87d4d383cb45031cb9fa2fc8ddca73fd6649240f` |
| `timm/res2net50_26w_4s.in1k` | `a36466d0354c9d337dbc0af329e13cb1cd7137fe` |
| `timm/senet154.gluon_in1k` | `86f6d239ed30acbe3aca0cbb5769da2621a4e02c` |
| `timm/seresnet50.a1_in1k` | `23f3482ee9acc4f51a2668a7cdf255fbd9420417` |
| `timm/vgg16.tv_in1k` | `b8d8aa2dd860af9233c8c67385a8097fd6c35d3f` |
| `timm/xception41.tf_in1k` | `8a17189361e63c972815ef62f2a30dd5b9f393b1` |

### Not Run / Remaining Gaps

- No Multi-Device, full Nightly, complete package or all-profile GPU qualification.
  DPN92, NFNet F3, Res2Net50d and Res2NeXt50 remain declared but were not GPU-selected.
- No latency equivalence or improvement claim. Reference timing still excludes
  preprocessing while the native Task includes family preprocessing; this existing
  accounting difference is documented, not declared fixed.
- Remote Pre-merge is pending at submission.

## Contributor Self-Review

- [x] I have completed a self-review of this change.

Reviewed ownership, original computation/preprocessing and validation criteria,
complete output metadata, consumer wiring and benchmark timing boundaries.

## Notes For Future Readers

Review one family from support/builder metadata through runtime binding and its
CPU/E2E tests. The other families intentionally duplicate owned code rather than
depending on each other. Use existing `TRTMC_NATIVE_BUILD_DIR` for development
consumer binaries. Runtime and family DSOs upgrade together; old bundles must
be rebuilt. This branch does not depend on another migration batch.

### Risk level

- [ ] Low
- [x] Medium
- [ ] High

Ten family Task identities change, but their computation and acceptance criteria
are preserved and each selected official model was exercised end to end.

### CODE REVIEW

**Comment by yifeif-nv** [comment]
Fixed in `a9221a6e27d0e59ac56e895a7aa41fada8442a8f` across all ten family-owned constructors. Supplied blank labels are rejected at load time when the vocabulary is unknown; absent labels, fully named labels and explicit vocabulary identities remain supported. Added positive and negative CPU regressions: the negative case failed before the fix, and all twenty family contract/preprocessing tests now pass. Each owner also has dependency-free support-discovery tests required by current main (328 owner Python tests passed). Model math, preprocessing and original numerical acceptance rules are unchanged.

**Comment by yifeif-nv** [comment]
The remaining build-time finding is fixed in `9d4a8b2658684a3f78cb6025bf1aefa774d2fa08`. All ten families now validate existing class metadata before engine construction. Eight builders do so before opening weights; MNASNet and VGG reuse their existing metadata extraction after weight loading but before building the engine. The regression failed on the old order and now verifies that invalid metadata never calls the engine builder. All 330 owner Python tests pass, with 14 unselected E2E cases skipped. Validation rules, model computation and acceptance thresholds are unchanged.
